crates/swc_ecma_transforms_base/src/fixer.rs

```diff
@@ -36,7 +37,7 @@ struct Fixer<'a> {
     ///
     /// Key is span of inner expression, and value is span of the paren
     /// expression.
-    span_map: AHashMap<Span, Span>,
+    span_map: FxHashMap<Span, Span>,
```

Comment:
I think we should use `IndexMap`. Iteration order of `fxhash` is deterministic but it's different from it of a vector.

Reply:
I expect get the same result from any order of iteration.

for example.

```JavaScript 
// span1
(
    // span2
    (
        // span3
        expr
    )
)
```

We will remove the parens and merge all comments. 
And we get the map

```
span_map {
    span2 -> span1,
    span3 -> span2,
}
```

Whether you get [`span1 span2`, `span3`] and then `span1 span2 span3` , or [`span1`, `span2 span3`] and then  `span1 span2 span3`, they should be same.

So FxHashMap should be fine.

But I don't know why AHashMap doesn't work, maybe it will lose data in some cases.

Reply:
Oh you are right, thanks